fix(edit-content): update tag field component and related styles

oidacra opened the pull request:
Improves tag field implementation with new component files and updates to form styles. Removes unused select item directives from UI library.

Refs: #30405

This pull request includes various changes to the `core-web` library, focusing on improvements to autocomplete functionality, form styling, and the removal of unused directives. The most important changes include refactoring the `dot-edit-content-tag-field` component, updating SCSS for form elements, and removing the `dot-select-item` directive.

Improvements to `dot-edit-content-tag-field` component:

* Refactored `dot-edit-content-tag-field.component.ts` to improve tag search functionality, including debouncing user input and error handling.
* Updated the HTML template to use PrimeNG's `AutoComplete` component with new attributes for better user experience.
* Added a new SCSS rule to ensure the component displays correctly.

Styling updates:

* Changed the display property from `grid` to `flex` in `_autocomplete.scss` to improve layout and removed unnecessary grid-template-columns rules.
* Updated the `#field-chip` styling in `common.scss` to use a new background color and reverse the flex direction.

Removal of unused directives:

* Removed the `dot-select-item` directive and its associated test file, as it is no longer needed. [[1]](diffhunk://#diff-bd390158a0315505d598e7ff896c73589c38f5263e0ff170662313807c45e15eL41) [[2]](diffhunk://#diff-446767b091d6c9eecf9386100c80ee201deb41a35084300ed7c19dec48fdf77aL1-L60) [[3]](diffhunk://#diff-edd6e21abd7366801d6a502155172a48d776818c1056bf0d8d393d30b6117cd3L1-L28)


https://github.com/user-attachments/assets/dab2af16-3d8d-4ab7-9a5b-146904b7bfa3


This PR fixes: #30405

oidacra committed:
fix(edit-content): update tag field component and related styles

Improves tag field implementation with new component files and updates to
form styles. Removes unused select item directives from UI library.

Refs: #30405

oidacra committed:
fix(edit-content): improve tag field component behavior and style classes dialog and add test to edit-content-tag-field

oidacra committed:
fix(template): fix heigth button in autocomplete

oidacra committed:
refactor(tag-field): replace hardcoded values with constants for minimum length and delay

Updated the tag field component to use constants for  and  instead of hardcoded values, improving maintainability and readability of the code.

nicobytes commented on core-web/libs/ui/src/lib/directives/dot-select-item/dot-select-item.directive.ts:
```diff
@@ -1,28 +0,0 @@
-    selector: 'p-autoComplete[dotSelectItem]',
-    standalone: true
-})
-export class DotSelectItemDirective {
```
This directive is used in other parts, are we sure to delete it?

oidacra replied:
Yes, I took a look but I can double-check.

oidacra committed:
fix(templates): implement unique class selection logic

Updated the add-style-classes-dialog component to use a constant for unique class selection, ensuring that duplicate classes cannot be added when the unique option is enabled. This change enhances the component's functionality and improves user experience.

oidacra committed:
fix(edit-content-sidebar): update API endpoint for contentlet retrieval

Changed the computed property  in the DotEditContentSidebarInformationComponent to use the new API versioning format, updating the URL from  to  for improved consistency and maintainability.

oidacra committed:
fix(templates) fix lint failing

oidacra committed:
Merge branch 'main' into issue-30405-tag-field-fix
